### PR TITLE
Add CI bot to bump golang version to match Kubernetes golang

### PR DESCRIPTION
**Pull request by medyagh**
- Also fixed the regex that was wrong
- Also changed the source of golang to KubeCross in kubernetes

### CODE REVIEW

File: .github/workflows/update-golang-version.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  workflow_dispatch:
+  schedule:
+    # every Monday at around 2 am pacific/9 am UTC
+    - cron: "0 9 * * 1"
+env:
+  GOPROXY: https://proxy.golang.org
+  GO_VERSION: 1.16.4
+jobs:
+  bump-k8s-versions:
+    runs-on: ubuntu-20.04
```
Comment (spowelljr) [inline .github/workflows/update-golang-version.yml:12]: Using `ubuntu-latest` will prevent us from having to bump Ubuntu version in the future.

File: .github/workflows/update-golang-version.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-go@v2
+        with:
+          go-version: ${{env.GO_VERSION}}
+          stable: true
+      - name: Bump Golang Versions
+        id: bumGolang
```
Comment (spowelljr) [inline .github/workflows/update-golang-version.yml:20]: Typo

File: hack/jenkins/common.sh [hunk 3]
```diff
@@ -91,7 +91,7 @@ else
 fi
 
 # installing golang so we could do go get for gopogh
-./installers/check_install_golang.sh "1.16.4" "/usr/local" || true
+./installers/check_install_golang.sh "1.16.6" "/usr/local" || true
```
Comment (spowelljr) [inline hack/jenkins/common.sh:94]: This is still `1.16.6`, is the regex working for this file?

Comment (medyagh) [reply hack/jenkins/common.sh:94]: good catch !

File: Makefile [hunk 4]
```diff
@@ -32,7 +32,7 @@ RPM_VERSION ?= $(DEB_VERSION)
 RPM_REVISION ?= 0
 
 # used by hack/jenkins/release_build_and_upload.sh and KVM_BUILD_IMAGE, see also BUILD_IMAGE below
-GO_VERSION ?= 1.16.6
+GO_VERSION ?= 1.16.4
```
Comment (spowelljr) [inline Makefile]: Are you bumping down the version to test the bot?

**Review by spowelljr — APPROVED** [review]

**Pull request merged**